sphinx/util/requests.py

```diff
@@ -50,6 +62,23 @@ def head(url: str, **kwargs: Any) -> requests.Response:
 
 
 class _Session(requests.Session):
+    def get_redirect_target(self, resp: requests.Response) -> str | None:
+        """Overrides the default requests.Session.get_redirect_target"""
+        # do not follow redirections that match ignored URI patterns
+        if resp.is_redirect:
+            location = resp.headers['location']
+            dest = urljoin(resp.url, location)
```

Comment:
I'd draw attention to this line in particular as possibly-dubious.

Servers often respond with relative `Location` redirects; but I think we want to match against more-fully-qualified URI paths.

Reply:
This looks good to me.